File: typescript/docs_code_snippets/functions-stream.ts
import { z } from "zod";
import { Opper } from "opperai";

const opper = new Opper();

// --- docs ---
for await (const chunk of opper.stream("docs-snippet-stream", {
  input: { text: "Explain streaming" },
  output_schema: z.object({ summary: z.string() }),
})) {
  if (chunk.type === "content") process.stdout.write(chunk.delta);
  if (chunk.type === "done") console.log();
}
// --- /docs ---

// cleanup
await opper.functions.delete("docs-snippet-stream");

File: python/docs_code_snippets/functions-stream.py
import sys
import time
from opperai import Opper

opper = Opper()

fn_name = f"docs-snippet-{int(time.time())}"

# --- docs ---
for chunk in opper.stream(
    fn_name,
    input={"text": "Explain streaming"},
    output_schema={"type": "object", "properties": {"summary": {"type": "string"}}},
):
    if chunk.type == "content":
        sys.stdout.write(chunk.delta)
        sys.stdout.flush()
    elif chunk.type == "done":
        print()
# --- /docs ---

# Cleanup
opper.functions.delete(fn_name)
